v2/pkg/subscraping/sources/quake/quake.go

```diff
@@ -97,6 +98,51 @@ func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Se
 				s.results++
 			}
 		}
+		pages = int(response.Meta.Pagination.Total/500) + 1
+		if pages > 1 {
+			for currentPage := 2; currentPage <= pages; currentPage++ {
+				var start = (currentPage - 1) * 500
+				requestBody = []byte(fmt.Sprintf(`{"query":"domain: %s", "include":["service.http.host"], "latest": true, "start":%d, "size":500 ,"latest":true}`, domain, start))
+				resp, err = session.Post(ctx, "https://quake.360.net/api/v3/search/quake_service", "", map[string]string{
```

Comment:
If we move the pagination logic to the outer scope, we can accomplish this in just one request. 

```go

pageSize := 500
		start := 0
		totalProcessed := 0

		for {
		
		....
		
		
		
		# at the end of loop something like this
		totalProcessed += len(response.Data)
			if totalProcessed >= response.Meta.Pagination.Total {
				break
			}

			start += pageSize
			...
```